### PR TITLE
Changes for Overlapping Timeslots , Token Validation & RetryLogic

### PR DESCRIPTION
**Pull request by rdeshmukh15**
# Description of change
- Added changes for API token validation before executing code in discover mode
- Made 'page_size' attribute configurable
- Increased timeslot boundaries by 1 ms for each interval
- Related Tickets: TDL-19621,TDL-19715

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_dixa/streams/conversations.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Conversations(IncrementalStream):
     endpoint = "/v1/conversation_export"
 
     # pylint: disable=signature-differs
-    def get_records(self, start_date):
+    def get_records(self, start_date, config: dict = {}):
```
Comment (kethan1122) [inline tap_dixa/streams/conversations.py:22]: is the config param being used in the method? If not, Is it ok to remove

Comment (Vi6hal) [reply tap_dixa/streams/conversations.py:22]: yes @kethan1122  this config is required for the page_size attribute which is used for the ActivityLog extraction

**Comment by kethan1122** [comment]
looks good to me

**Review by kethan1122 — APPROVED** [review]

File: tap_dixa/discover.py [hunk 2]
```diff
@@ -1,17 +1,17 @@
 """ Module providing disovery method of tap-dixa"""
 import json
-
 from singer import metadata
 from singer.catalog import Catalog
-from tap_dixa.streams import STREAMS
-
+from .streams import STREAMS, ActivityLogs
+from .client import Client
```
[the remark this answers is not in the document]
Comment (Vi6hal) [reply tap_dixa/discover.py]: done

File: tap_dixa/streams/messages.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class Messages(IncrementalStream):
     endpoint = "/v1/message_export"
 
     # pylint: disable=signature-differs
-    def get_records(self, start_date):
+    def get_records(self, start_date, config: dict = {}):
```
[the remark this answers is not in the document]
Comment (Vi6hal) [reply tap_dixa/streams/messages.py:22]: https://github.com/singer-io/tap-dixa/pull/9/#discussion_r914458369
FYR

**Pull request merged**